pkg/tbtc/chain.go

```diff
@@ -26,6 +29,15 @@ type DistributedKeyGenerationChain interface {
 	OnDKGStarted(
 		func(event *DKGStartedEvent),
 	) subscription.EventSubscription
+	OnDKGResultSubmitted(
+		func(event *DKGResultSubmittedEvent),
+	) subscription.EventSubscription
+	SubmitResult(
```

Comment:
The `dkg.ResultSubmitter` and `dkg.ResultSigner` should not be directly implemented in the `pkg/chain/ethereum` as we also have application-specific logic that should be implemented in `pkg/tbtc`. Here is what I propose:

1. In `node.go`, create two package-private structs `dkgResultSubmitter` and `dkgResultSigner` that will implement `dkg.ResultSubmitter` and `dkg.ResultSigner` respectively. Both should have references to the`tbtc.Chain`
2. The aforementioned types should do all the app-specific actions like submission eligibility and so on, and then call respective functions of the `tbtc.Chain`. Chain functions should be as small as possible and focus only on things required to interact with the contract. Nothing else. In other words:

    - `dkgResultSubmitter` should call chain's `OnDKGResultSubmitted` and `SubmitDKGResult`
    - `dkgResultSigner` should call functions of the signer returned by chain's `Signing` function and the `CalculateDKGResultHash` function.
3. Add a `TODO` above `node.joinDKGIfEligible` about extracting DKG-specific code into a separate file `pkg/tbtc/dkg.go`. I'll address it once #3169 is merged as the DKG stuff become really big and will clutter the `node.go` file quickly.

Reply:
Done in 655ee9ba9b2f7e587a16d5c119fd843909339517.